src/gt4py/backend/pyext_builder.py

```diff
@@ -57,6 +57,7 @@ def get_gt_pyext_build_opts(
             "-DBOOST_OPTIONAL_CONFIG_USE_OLD_IMPLEMENTATION_OF_OPTIONAL",
             "-DBOOST_OPTIONAL_USE_OLD_DEFINITION_OF_NONE",
             "--expt-relaxed-constexpr",
+            "--disable-warnings",  # workaround for 'catastrophic failure' error in nvcc
```

Comment:
You mean that users should individually set it if needed?

I'm not sure if compiler warnings really add anything in general, it just seems to generate a lot of output really, so we might as well just always mute them.

Reply:
I agree about the muting compiler warnings by default, but instead of hardcoding it in the source code, I'd rather do it by adding the `--disable-warnings` flag to the default list of nvcc compiler flags, so the user could potentially turn it off if he needs to. Something like:

```
    "extra_compile_args": {"cxx": [], "nvcc": ["--disable-warnings"]},   # "--disable-warnings" prevents 'catastrophic failure' error in old (<11) nvcc versions 
```